internal/pkg/cmd/reloader.go

```diff
@@ -246,6 +248,11 @@ func getNamespaceLabelSelector(cmd *cobra.Command) (string, error) {
 		logrus.Fatal(err)
 	}
 
+	if !isGlobal && len(namespaceLabelSelector) > 0 {
```

Comment:
shouldn't it be `if isGlobal && len(namespaceLabelSelector) > 0`?

Reply:
If the reloader isn't watch globally, the reloader should have no permission to get/list/filter the namespace. Because of that, the reloader should ignore the setting of the selector.